controllers/component_controller.go

```diff
@@ -47,16 +56,105 @@ type ComponentReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
 func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling Component")
 
-	// TODO(user): your logic here
+	// Fetch the Component instance
+	instance := &corev1alpha1.Component{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
 
+	if !instance.DeletionTimestamp.IsZero() {
+		// The object is being deleted.
+		logger.Info("Component is being deleted")
+
+		// Remove finalizer to complete the deletion.
+		instance.Finalizers = utils.RemoveString(instance.Finalizers, corev1alpha1.ComponentFinalizer)
+		if err := r.Client.Update(ctx, instance); err != nil {
+			logger.Error(err, "Failed to remove component finalizer")
+		}
+		return reconcile.Result{}, err
+	}
+
+	requeue, err := r.UpdateComponent(ctx, logger, instance)
```

Comment:
Refer to the latest processing logic of the repository and don't let the update event and reconcile occur at the same time.